Update django-two-factor-auth and django-otp

millerdev opened the pull request:
https://github.com/dimagi/commcare-hq/commit/3b4b73ae43ab4a8a7a2fb95132b73510ec879445 breaks the deploy script, so cannot be deployed to staging. It also [breaks tests](https://travis-ci.org/github/dimagi/commcare-hq/builds/707634562).

Note: this is merging into dm/dj2-fix-tests, not master. dm/dj2-fix-tests is in flux right now, and likely will be rebased or split up into a set of smaller PRs eventually.

~TODO fix the deploy and tests~ Fixed in https://github.com/dimagi/commcare-hq/pull/28095/commits/c6c7abc4085c9d00491ad2a13a86f66f84aa8aeb

millerdev committed:
Update django-two-factor-auth and django-otp

dannyroberts commented:
The test failure when deploying to staging (also present in tests I believe) is
```
10.201.11.173] out: ERRORS:
[10.201.11.173] out: ?: (urls.E004) Your URL pattern 'two_factor' is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.
[10.201.11.173] out: 	HINT: Try removing the string 'two_factor'. The list of urlpatterns should not have a prefix string as the first element.
[10.201.11.173] out: ?: (urls.E004) Your URL pattern 'two_factor_twilio' is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.
[10.201.11.173] out: 	HINT: Try removing the string 'two_factor_twilio'. The list of urlpatterns should not have a prefix string as the first element.
[10.201.11.173] out: ?: (urls.E004) Your URL pattern [<URLPattern '^account/login/$' [name='login']>, <URLPattern '^account/two_factor/setup/$' [name='setup']>, <URLPattern '^account/two_factor/qrcode/$' [name='qr']>, <URLPattern '^account/two_factor/setup/complete/$' [name='setup_complete']>, <URLPattern '^account/two_factor/backup/tokens/$' [name='backup_tokens']>, <URLPattern '^account/two_factor/backup/phone/register/$' [name='phone_create']>, <URLPattern '^account/two_factor/backup/phone/unregister/(?P<pk>\d+)/$' [name='phone_delete']>, <URLPattern '^account/two_factor/$' [name='profile']>, <URLPattern '^account/two_factor/disable/$' [name='disable']>] is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.
[10.201.11.173] out: ?: (urls.E004) Your URL pattern [<URLPattern '^twilio/inbound/two_factor/(?P<token>\d+)/$' [name='call_app']>] is invalid. Ensure that urlpatterns is a list of path() and/or re_path()
```

millerdev committed:
Handle django-two-factor-auth urlpatterns tuples

Fixes error on ./manage.py runserver:

SystemCheckError: System check identified some issues:

ERRORS:
?: (urls.E004) Your URL pattern 'two_factor' is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.
	HINT: Try removing the string 'two_factor'. The list of urlpatterns should not have a prefix string as the first element.
?: (urls.E004) Your URL pattern 'two_factor_twilio' is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.
	HINT: Try removing the string 'two_factor_twilio'. The list of urlpatterns should not have a prefix string as the first element.
?: (urls.E004) Your URL pattern [<URLPattern '^account/login/$' [name='login']>, <URLPattern '^account/two_factor/setup/$' [name='setup']>, <URLPattern '^account/two_factor/qrcode/$' [name='qr']>, <URLPattern '^account/two_factor/setup/complete/$' [name='setup_complete']>, <URLPattern '^account/two_factor/backup/tokens/$' [name='backup_tokens']>, <URLPattern '^account/two_factor/backup/phone/register/$' [name='phone_create']>, <URLPattern '^account/two_factor/backup/phone/unregister/(?P<pk>\d+)/$' [name='phone_delete']>, <URLPattern '^account/two_factor/$' [name='profile']>, <URLPattern '^account/two_factor/disable/$' [name='disable']>] is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.
?: (urls.E004) Your URL pattern [<URLPattern '^twilio/inbound/two_factor/(?P<token>\d+)/$' [name='call_app']>] is invalid. Ensure that urlpatterns is a list of path() and/or re_path() instances.

millerdev commented:
I'm going to cherry-pick this onto a different branch. Closing to avoid unnecessary merge commits.
